.github/workflows/ingestion-cleanup-deploy.yml

```diff
@@ -0,0 +1,60 @@
+name: Ingestion cleanup deploy
+
+on:
+  push:
+    branches: [main, '*-stable']
+    paths:
+    - '.github/workflows/ingestion-cleanup-deploy.yml'
+    - 'ingestion/functions/**'
+    - '!ingestion/functions/README.md'
```

Comment:
I'd add `!ingestion/functions/parsing/**` and `!ingestion/functions/retrieval/**` to remove unnecessary builds